### PR TITLE
Pretty print

### PR DESCRIPTION
**Pull request by j-hui**
not complete yet, but ¯\_(ツ)_/¯ 

if we have a complete frontend, that's at least something to s how for ourselves too at the demo

### CODE REVIEW

**Comment by j-hui** [comment]
do not merge yet

File: src/frontend/astprint.ml [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
   | Int -> "int"
   | Float -> "float"
   | String -> "string"
   | Bool -> "bool" 
-  | Struct t -> "struct" (* we want to separate this later for type checking for structs *) 
-  | Array t -> "array" (*same here *)  
-  | Vector t -> "vector"
+  | Struct t -> "struct " ^ t 
+  | Array t -> (_string_of_typ t) ^ "[]"
+  | Vector t -> "vector<" ^ string_of_int(t) ^ ">"
```
Comment (Contentmaudlin) [inline src/frontend/astprint.ml:18]: "<" and ">" is not our vector notation anymore right?

Comment (j-hui) [reply src/frontend/astprint.ml:18]: for types it's still `vector<n>`

File: src/frontend/astprint.ml [hunk 2]
```diff
@@ -60,54 +60,58 @@ let string_of_unop = function
 
 let string_of_mut = function
 	| Immutable -> ""
-	| Mutable -> "var"
+	| Mutable -> "var "
 
-let string_of_lit = function
+
+let string_of_list f l s = String.concat s (List.map f l)
```
Comment (Contentmaudlin) [inline src/frontend/astprint.ml:66]: Ya I should have done this lol

Comment (j-hui) [reply src/frontend/astprint.ml:66]: there are a couple of other Neat Tricks™ I'm looking to introduce, including nice indentation

so _actually_ **pretty** print

**Review by Contentmaudlin — APPROVED** [review]
LGTM

**Pull request merged**